Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -166,6 +182,91 @@ public final class KafkaConsumer: Sendable, Service {
+    ///
+    /// - Parameters:
+    ///     - config: The ``KafkaConsumerConfiguration`` for configuring the ``KafkaConsumer``.
+    ///     - logger: A logger.
+    /// - Returns: The newly created ``KafkaConsumer``.
+    /// - Throws: A ``KafkaError`` if the initialization failed.
+    public convenience init(
+        config: KafkaConsumerConfiguration,
+        logger: Logger
+    ) throws {
+        let stateMachine = NIOLockedValueBox(StateMachine(logger: logger))
+
+        var subscribedEvents: [RDKafkaEvent] = [.log, .fetch]
+        // Only listen to offset commit events when autoCommit is false
+        if config.enableAutoCommit == false {
+            subscribedEvents.append(.offsetCommit)
+        }
+
+        let client = try RDKafkaClient.makeClient(
+            type: .consumer,
+            configDictionary: config.dictionary,
+            events: subscribedEvents,
+            logger: logger
+        )
+
+        try self.init(
+            client: client,
+            stateMachine: stateMachine,
+            config: config,
+            logger: logger
+        )
+    }
+
+    /// Initialize a new ``KafkaConsumer`` and a ``KafkaConsumerEvents`` asynchronous sequence.
+    ///
+    /// Use the asynchronous sequence to consume events.
+    ///
+    /// - Important: When the asynchronous sequence is deinited the producer will be shutdown and disallow sending more messages.
+    /// Additionally, make sure to consume the asynchronous sequence otherwise the events will be buffered in memory indefinitely.
+    ///
+    /// - Parameters:
+    ///     - config: The ``KafkaConsumerConfiguration`` for configuring the ``KafkaConsumer``.
+    ///     - logger: A logger.
+    /// - Returns: A tuple containing the created ``KafkaConsumer`` and the ``KafkaConsumerEvents``
+    /// `AsyncSequence` used for receiving message events.
+    /// - Throws: A ``KafkaError`` if the initialization failed.
+    public static func makeConsumerWithEvents(
+        config: KafkaConsumerConfiguration,
+        logger: Logger
+    ) throws -> (KafkaConsumer, KafkaConsumerEvents) {
+        let stateMachine = NIOLockedValueBox(StateMachine(logger: logger))
+
+        var subscribedEvents: [RDKafkaEvent] = [.log, .fetch]
+        // Only listen to offset commit events when autoCommit is false
+        if config.enableAutoCommit == false {
+            subscribedEvents.append(.offsetCommit)
+        }
+
+        let client = try RDKafkaClient.makeClient(
+            type: .consumer,
+            configDictionary: config.dictionary,
+            events: subscribedEvents,
+            logger: logger
+        )
+
+        let consumer = try KafkaConsumer(
+            client: client,
+            stateMachine: stateMachine,
+            config: config,
+            logger: logger
+        )
+
+        let sourceAndSequence = NIOAsyncSequenceProducer.makeSequence(
```

Comment:
I am not sure I fully understand an idea here: it seems like nothing is supposed to be supplied in this source...
Shouldn't be `sourceAndSequence.source` assigned anywhere?

Reply:
Yes, the primary intention of this PR was to expose the `KafkaConsumerEvents` itself without serving any events at the moment.

Following onto that PR events like rebalance etc. can be easily added into that existing `KafkaConsumerEvents` sequence.